packages/widget/src/AppRoutes.tsx

```diff
@@ -119,6 +122,12 @@ const routes: RouteObject[] = [
 ];
 
 export const AppRoutes = () => {
+  const emitter = useWidgetEvents();
   const element = useRoutes(routes);
+  const location = useLocation();
+
+  useEffect(() => {
+    emitter.emit(WidgetEvent.Page, location.pathname);
+  }, [emitter, location.pathname]);
```

Comment:
Instead of making this component (the whole widget?) to re-render every time we navigate to a new page, let create a wrapper around `useNavigate` hook and emit event on user action. Basically, we can create a custom `useNavigate` and use react-router one inside together with an even emitter.

Reply:
Nice idea!